restcomm.android.sdk/src/main/java/org/restcomm/android/sdk/RCDevice.java

```diff
@@ -1059,15 +1010,21 @@ public void onOpenReply(String jobId, RCDeviceListener.RCConnectivityStatus conn
          }
          else {
             RCLogger.w(TAG, "RCDeviceListener event suppressed since Restcomm Client Service not attached: onInitialized(): " +
-                  RCClient.errorText(status));
+                    RCClient.errorText(status));
          }
          return;
       }
 
-      //we dont need to clear data
-      registerForPush(clearPushData);
-      clearPushData = false;
-
+      state = DeviceState.READY;
+      if (isServiceAttached) {
```

Comment:
This whole if leg seems a duplicate of the one previous; do we need it? Am I missing sth?

Reply:
I don't follow on this one.